store/src/java/com/zimbra/cs/html/owasp/OwaspHtmlSanitizer.java

```diff
@@ -38,10 +39,38 @@ public class OwaspHtmlSanitizer implements Callable<String> {
     private String vHost;
 
     public OwaspHtmlSanitizer(String html, boolean neuterImages, String vHost) {
-        this.html = html;
+    	// fix to check extra tags & remove
+        this.html = checkUnbalancedTags(html);;
```

Comment:
Before sanitize() only we need to call this one . as sanitize() is failing to remove unbalanced tags . so we are removing & then doing sanitization

Reply:
If my understanding is correct, html is loaded and sanitized around here.
```
        // create a thread-specific policy
        instantiatePolicy();
        final Policy policy = POLICY_DEFINITION.apply(new StyleTagReceiver(renderer));
        // run the html through the sanitizer
        HtmlSanitizer.sanitize(html, policy);
```
So can we put `checkUnbalancedTags` after null check (just before sanitizing)?
```
        if (StringUtil.isNullOrEmpty(html)) {
            return null;
        }
```
Otherwise, if `checkUnbalancedTags` is called at line 43 without null check and if html is null, NullPointerException is thrown, I think.

Reply:
@k-kato right, should add a null check. @rupeshtelus please add a null check as suggested above.

Reply:
added null validation